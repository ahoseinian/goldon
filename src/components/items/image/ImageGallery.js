import React, {Component, PropTypes} from 'react'

class ImageGallery extends Component {
  static propTypes = {
    images: PropTypes.array.isRequired
  }

  render() {
    return (
      <div>ImageGallery</div>
    )
  }

}

export default ImageGallery
