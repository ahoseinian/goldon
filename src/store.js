import {createStore, applyMiddleware} from 'redux'
import goldonApp from './reducers'
// import createLogger from 'redux-logger'
import createPromiseMiddleware from 'redux-promise-middleware'

// const logger = createLogger({collapsed: true});
const promiseMiddleware = createPromiseMiddleware();
const store = createStore(goldonApp, applyMiddleware(promiseMiddleware))


export default store
