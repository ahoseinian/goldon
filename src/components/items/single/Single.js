import React from 'react';
import {connect} from 'react-redux'
import styled from 'styled-components'
import {Helmet} from 'react-helmet'
import ImageGallery from '../image/ImageGallery'
import GalleryModal from '../../utils/GalleryModal'
import findById from '../../../reducers/items/findById'
import similarItems from '../../../reducers/items/similarItems'
import {showInGallery} from '../../../actions'
import Details from './Details'
import MainInfo from './MainInfo'
import List from '../list'
import SimpleCard from '../../common/card/SimpleCard'

const Wrapper = styled.div `
  margin-top: 4.5rem;
  margin-bottom: 1rem;
`

export class Single extends React.Component {
  static propTypes = {
    params: React.PropTypes.shape({
      id: React.PropTypes.string.isRequired
    }),
    item: React.PropTypes.object,
    handleZoom: React.PropTypes.func
  }

  componentWillMount(){
    window.scrollTo(0, 0)
  }

  componentWillReceiveProps(nextProps) {
    window.scrollTo(0, 0)
  }

  render() {
    const {item, similarItems, showInGallery} = this.props
    return !item ? null :(
      <Wrapper className="container">
        <GalleryModal />
        <Helmet>
          <title>{item.name}</title>
        </Helmet>
        <div className="row">
          <div className="col-sm-4">
            <ImageGallery images={item.images} handleZoom={showInGallery}/>
          </div>
          <div className="col-sm-8 mt-2">
            <MainInfo item={item}/>
          </div>
        </div>
        <div className="row ">
          <div className="col-12">
            <Details item={item}/>
            {
              similarItems &&
              <div className="similar-items">
                <SimpleCard title="اقلام مشابه">
                  <List items={similarItems}/>
                </SimpleCard>
              </div>
            }
          </div>
        </div>
      </Wrapper>
    )
  }
}

export const mapStateToProps = (state, {params}) => ({
  item: findById(state.items, params.id),
  similarItems: similarItems(state.items, params.id)
})

export const mapDispatchToProps = { showInGallery }

export default connect(mapStateToProps, mapDispatchToProps)(Single);
