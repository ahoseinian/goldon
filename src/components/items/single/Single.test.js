import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import ImageGallery from '../image/ImageGallery'
import {Helmet} from 'react-helmet'
import {Single, mapStateToProps, mapDispatchToProps} from './Single'
import store from '../../../store'

const item = {
  name: 'item name',
  images: []
}

describe('Single', () => {
  it('does not renders similarItems without them', () => {
    const wrapper = shallow(<Single item={item}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders similarItems', () => {
    const wrapper = shallow(<Single item={item} similarItems={[{}]}/>)
    expect(toJson(wrapper.find('.similar-items'))).toMatchSnapshot();
  })

  it('scrolls to zero position', () => {
    global.scrollTo = jest.fn()
    const wrapper = shallow(<Single item={item} similarItems={[{}]}/>)
    expect(global.scrollTo.mock.calls[0]).toEqual([0, 0])
    wrapper.setProps({
      item: {
        ...item,
        name: 'akbar'
      }
    })
    expect(global.scrollTo.mock.calls[1]).toEqual([0, 0])
  })

  it('changes document title', () => {
    const wrapper = shallow(<Single item={item} similarItems={[{}]}/>)
    expect(wrapper.find(Helmet).find('title').text()).toEqual('item name')
  })

  it('paths handleZoom prop to ImageGallery', () => {
    const handleZoomFunc = jest.fn()
    const wrapper = shallow(<Single item={item} similarItems={[{}]} handleZoom={handleZoomFunc}/>)
    expect(wrapper.find(ImageGallery).props().handleZoom).toEqual(handleZoomFunc)
  })
})

describe('mapStateToProps', () => {
  it('returns item by id', () => {
    const state = {
      items: [
        { id: 'test' },
        { id: 'test3' }
      ]
    }
    expect(
      mapStateToProps(state, {
        params: { id: 'test3' }
      }).item
    ).toEqual({id: 'test3'})
  })
})


describe('mapDispatchToProps', () => {
  it('returns showInGallery function', () => {
    expect(typeof mapDispatchToProps.showInGallery).toEqual('function')
  })
});
