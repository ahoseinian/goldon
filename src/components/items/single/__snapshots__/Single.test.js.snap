exports[`Single does not renders similarItems without them 1`] = `
<styled.div
  className="container">
  <Connect(Component) />
  <HelmetWrapper
    encodeSpecialCharacters={true}>
    <title>
      item name
    </title>
  </HelmetWrapper>
  <div
    className="row">
    <div
      className="col-sm-4">
      <ImageGallery
        images={Array []} />
    </div>
    <div
      className="col-sm-8 mt-2">
      <MainInfo
        item={
          Object {
            "images": Array [],
            "name": "item name",
          }
        } />
    </div>
  </div>
  <div
    className="row ">
    <div
      className="col-12">
      <Details
        item={
          Object {
            "images": Array [],
            "name": "item name",
          }
        } />
    </div>
  </div>
</styled.div>
`;

exports[`Single renders similarItems 1`] = `
<div
  className="similar-items">
  <SimpleCard
    title="اقلام مشابه">
    <List
      items={
        Array [
          Object {},
        ]
      } />
  </SimpleCard>
</div>
`;
